cluster-autoscaler/clusterstate/api/types.go

```diff
@@ -69,36 +64,135 @@ const (
 )
 
-// ClusterAutoscalerCondition describes some aspect of ClusterAutoscaler work.
-type ClusterAutoscalerCondition struct {
-	// Type defines the aspect that the condition describes. For example, it can be Health or ScaleUp/Down activity.
-	Type ClusterAutoscalerConditionType `json:"type,omitempty"`
-	// Status of the condition.
-	Status ClusterAutoscalerConditionStatus `json:"status,omitempty"`
-	// Message is a free text extra information about the condition. It may contain some
-	// extra debugging data, like why the cluster is unhealthy.
-	Message string `json:"message,omitempty"`
-	// Reason is a unique, one-word, CamelCase reason for the condition's last transition.
-	Reason string `json:"reason,omitempty"`
+// RegisteredUnreadyNodeCount contains node counts of registered but unready nodes.
+type RegisteredUnreadyNodeCount struct {
+	// Total number of registered but unready nodes.
+	Total int `json:"total" yaml:"total"`
+	// ResourceUnready is the number of registered but unready nodes due to a missing resource (e.g. GPU).
+	ResourceUnready int `json:"resourceUnready" yaml:"resourceUnready"`
+}
+
+// RegisteredNodeCount contains node counts of registered nodes.
+type RegisteredNodeCount struct {
+	Total      int `json:"total" yaml:"total"`
+	Ready      int `json:"ready" yaml:"ready"`
+	NotStarted int `json:"notStarted" yaml:"notStarted"`
+	// Number of nodes that are being currently deleted. They exist in K8S but are not included in NodeGroup.TargetSize().
+	BeingDeleted int                        `json:"beingDeleted,omitempty" yaml:"beingDeleted,omitempty"`
+	Unready      RegisteredUnreadyNodeCount `json:"unready,omitempty" yaml:"unready,omitempty"`
+}
+
+// NodeCount contains number of nodes that satisfy different criteria.
+type NodeCount struct {
+	Registered       RegisteredNodeCount `json:"registered,omitempty" yaml:"registered,omitempty"`
+	LongUnregistered int                 `json:"longUnregistered" yaml:"longUnregistered"`
+	Unregistered     int                 `json:"unregistered" yaml:"unregistered"`
+}
+
+// ClusterHealthCondition contains information about health condition for the whole cluster.
+type ClusterHealthCondition struct {
+	// Status of cluster health.
+	Status ClusterAutoscalerConditionStatus `json:"status,omitempty" yaml:"status,omitempty"`
+	// NodeCounts contains number of nodes that satisfy different criteria in the cluster.
+	NodeCounts NodeCount `json:"nodeCounts,omitempty" yaml:"nodeCounts,omitempty"`
 	// LastProbeTime is the last time we probed the condition.
-	LastProbeTime metav1.Time `json:"lastProbeTime,omitempty"`
+	LastProbeTime metav1.Time `json:"lastProbeTime,omitempty" yaml:"lastProbeTime,omitempty"`
 	// LastTransitionTime is the time since when the condition was in the given state.
-	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty" yaml:"lastTransitionTime,omitempty"`
```

Comment:
Again, this is not a comment to this PR as some thought for potential follow-up: do we actually need those probe/transition times? What we used previously was essentially standard Kubernetes status conditions, so naturally they had a standard set of fields.

New structs look very differently from kube status conditions, so there is no point adhering to them as standard. And I'm thinking maybe those fields are not adding much value:
1. We define lastTransitionTime as time when Status field changed. But that seems rather arbitrary, given that we don't consider change in other fields as transition.
2. If we got rid of timestamps maybe we wouldn't need to write the configmap every loop? We could just write-on-change which should save constant writes to api-server in most clusters (admittedly that saves QPS when clusters are likely not under heavy load and doesn't help at all when nodes are being added - which is most likely when api-server is experiencing maximum load).
3. We could get rid of extra logic needed to handle those fields.

WDYT @x13n @walidghallab ?